torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3705,17 +3763,43 @@ struct to_ir {
             range, sv->asValue(val_range, method), subscript_exprs));
       }
     } else {
-      // Desugars gather syntactic sugar foo[i]
-      Value* idx = emitExpr(subscript_exprs[0]);
-      Value* val = sv->asValue(val_range, method);
       AT_ASSERT(subscript_exprs.size() == 1);
+      Value* sliceable = sv->asValue(val_range, method);
+
+      auto subscript_sv = emitSugaredExpr(subscript_exprs[0], 1);
```

Comment:
Consider adding a comment like the one above which explains that this code handles slice objects.

Reply:
Done.